dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AudioToText;
+using Microsoft.SemanticKernel.Contents;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.AssemblyAI;
+
+/// <summary>
+/// AssemblyAI speech-to-text service.
+/// </summary>
+public sealed class AssemblyAIAudioToTextService : IAudioToTextService
+{
+    private const string FallbackBaseUrl = "https://api.assemblyai.com/";
+    internal string ApiKey { get; }
+    internal HttpClient HttpClient { get; }
```

Comment:
Shouldn't the two properties - ApiKey and HttpClient - be private?

Reply:
I made them internal for unit testing purposes. 
The unit test verifies that the API key and http client configured through the extensions is the same as on the service.

Reply:
This is not good practise; bending code to unit testing. Change this properties to private.

Reply:
@Swimburger I have never used this before. But it may satisfy your requirement without making properties internal/public. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute?view=net-8.0

Reply:
@Krzysztof318 net8 feature, but all projects target netstandard2.0 or net6.
I don't mind targeting net8, or omitting the verification of API key and http client.

Reply:
Okay, I forgot we are still targeting net6.0 in tests projects.

Reply:
I changed them to private fields and removed the verification in unit tests.